src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,878 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
         private AssemblyLoadTestFixtureCollection() { }
     }
 
-    [Collection(AssemblyLoadTestFixtureCollection.Name)]
     public sealed class DefaultAnalyzerAssemblyLoaderTests : TestBase
     {
+#if NETCOREAPP
+        private sealed class InvokeUtil
+        {
+            public void Exec(string typeName, string methodName) => InvokeTestCode(typeName, methodName);
+        }
+#else
+        private sealed class InvokeUtil : MarshalByRefObject
+        {
+            public void Exec(string typeName, string methodName)
+            {
+                try
+                {
+                    InvokeTestCode(typeName, methodName);
+                }
+                catch (TargetInvocationException ex) when (ex.InnerException is XunitException)
+                {
+                    var inner = ex.InnerException;
+                    throw new Exception(inner.Message + inner.StackTrace);
+                }
+            }
+        }
+#endif
+
         private static readonly CSharpCompilationOptions s_dllWithMaxWarningLevel = new(OutputKind.DynamicallyLinkedLibrary, warningLevel: CodeAnalysis.Diagnostic.MaxWarningLevel);
         private readonly ITestOutputHelper _output;
-        private readonly AssemblyLoadTestFixture _testFixture;
-        public DefaultAnalyzerAssemblyLoaderTests(ITestOutputHelper output, AssemblyLoadTestFixture testFixture)
+
+        public DefaultAnalyzerAssemblyLoaderTests(ITestOutputHelper output)
         {
             _output = output;
-            _testFixture = testFixture;
+        }
+
+        private void Run(Action<DefaultAnalyzerAssemblyLoader, AssemblyLoadTestFixture> action, [CallerMemberName] string? memberName = null)
+        {
+#if NETCOREAPP
+            var alc = AssemblyLoadContextUtils.Create($"Test {memberName}");
```

Comment:
Could you explain what test gaps we are filling in by doing this indirection on the netcore side? I thought we already didn't have to worry about analyzer assemblies being loaded in any ALC which persists across tests.

Reply:
Tried to explain this a bit in the PR description. 

The gap is not as big for `AssemblyLoadContext` because, as you mention, the isolation is already there. There is no cross test pollution to worry about, not significantly at least. 

What this does buy us though is verifying how we handle the default ALC vs. the compiler ALC. Before this change in our tests the default ALC was the compiler ALC while now they are different instances. When they are different is the more complex case and how VS executes. This gives us better confidence we're falling back to the compiler ALC during loads vs. accidentally falling back to default.